src/components/molecules/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback, useMemo } from "react";
+import classNames from "classnames";
+import { format, fromUnixTime, getUnixTime } from "date-fns";
+import { range } from "lodash";
+import { useCss } from "react-use";
+
+import { RoomWithEvents, ScheduleProps } from "./Schedule.types";
+
+import { calcStartPosition } from "./Schedule.utils";
+
+import { ScheduleRoomEvents } from "../ScheduleRoomEvents";
+
+import "./Schedule.scss";
+
+const MAX_SCHEDULE_START_HOUR = 16;
+const MAX_HOUR = 24;
+const MIDDAY_HOUR = 12;
+
+export const Schedule: React.FC<ScheduleProps> = ({ scheduleDay }) => {
+  const getStartHour = useCallback(
+    (utcSeconds: number) => {
+      return utcSeconds >= scheduleDay.dayStartUtcSeconds
+        ? Number(format(fromUnixTime(utcSeconds), "H"))
+        : 0;
+    },
+    [scheduleDay]
+  );
+
+  const roomStartHour = useCallback(
+    (room: RoomWithEvents) =>
+      room.events?.reduce(
+        (acc, event) => Math.min(acc, getStartHour(event.start_utc_seconds)),
+        MAX_SCHEDULE_START_HOUR
+      ) ?? MAX_SCHEDULE_START_HOUR,
+    [getStartHour]
+  );
+
+  const scheduleStartHour = useMemo(
+    () =>
+      Math.min(
+        ...scheduleDay.rooms.map((room) => roomStartHour(room)),
+        MAX_SCHEDULE_START_HOUR
+      ),
+    [scheduleDay, roomStartHour]
+  );
+
+  const hours = useMemo(
+    () =>
+      range(scheduleStartHour, MAX_HOUR).map(
+        (hour: number) =>
+          `${hour % MIDDAY_HOUR || MIDDAY_HOUR} ${
+            hour >= MIDDAY_HOUR ? "PM" : "AM"
+          }`
+      ),
+    [scheduleStartHour]
+  );
+
+  const containerVars = useCss({
+    "--room-count": scheduleDay.rooms.length + 1,
+    "--current-time--position": calcStartPosition(
+      Math.floor(getUnixTime(Date.now())),
```

Comment:
@mike-lvov As per https://github.com/sparkletown/sparkle/pull/1302#discussion_r628966546, what are you suggesting is moved into a const?

Reply:
I haven't checked the logic in `calcStartPosition`, but since we're using `Date.now` in `calcStartPosition(Math.floor(getUnixTime(Date.now()))`, if the returned value changes a lot based on the current time, this will cause a re-render basically every time its called.

Reply:
used const `currentTimePosition`: https://github.com/sparkletown/sparkle/pull/1302/commits/3773b544eaa9b24083615b54b444167ca724eff2

`Date.now` is used to show current time line in the schedule (purple one). I think we do need to rerender it when the time is changed.
![image](https://user-images.githubusercontent.com/63313732/118136849-8e7d1c00-b3d2-11eb-9ad9-065fdbdcf28c.png)

`calcStartPosition` calculates left margin of the element based on `startTime` - used for events and current time line.

or do you have any better ideas how to show that current time line?

Reply:
> I think we do need to rerender it when the time is changed.

But not every time the component renders. This would be something that we would likely handle with a `useInterval` that updates some state every X timeframe (eg. minute) to trigger the re-render.

Reply:
done here: https://github.com/sparkletown/sparkle/pull/1302/commits/84c633a2848faab78da90654662b3d1b8aba1942